src/partition/repart.c

```diff
@@ -1126,7 +1126,7 @@ static uint64_t scale_by_weight(uint64_t value, uint64_t weight, uint64_t weight
         assert(weight_sum >= weight);
 
         for (;;) {
-                if (weight == 0)
+                if (IN_SET(weight, 0, UINT32_MAX))
```

Comment:
is UINT32_MAX ever seen here? can this happen? i mean you initialize the weight right after parsing the drop-in, if it is UINT32_MAX, hence this should never be seen?

Reply:
This was just to be safe, I can drop it again

Reply:
I guess with `--copy-from=` this could be triggered for example.

Reply:
Should --copy-from= initialize the field properly then?

sounds to me as if we shoudl assert() here, and insist that the field is initialized properly?